src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -300,7 +300,18 @@ await stream.CopyToAsync(httpConnection.OutputStream, 4096, cancellationToken)
 					}
 				}
 
-				statusCode = await Task.Run (() => (HttpStatusCode)httpConnection.ResponseCode).ConfigureAwait (false);
+				statusCode = await Task.Run(() => {
+					try {
+						return (HttpStatusCode)httpConnection.ResponseCode;
+					}
+					// Accessing the ResponseCode will throw an IOException if the code is 401
+					// and the server has not returned a WWW-Authenticate header on < API 19
+					catch (IOException ex) when (ex.Message.Contains("authentication challenges")) {
```

Comment:
I'm wondering how `IOException` compiles when both `System.IO` and `Java.IO` are in scope, and both of those namespaces have an `IOException` type...

I also find it somewhat amusing that the fix to `httpConnection.ResponseCode` throwing an exception...is to call it *again*.

> insanity is doing the same thing over and over and expecting a different result.

Oh Android, I ❤️ you so much...